### PR TITLE
fix(player): keep next-up video visible through episode transitions

### PR DESCRIPTION
**Pull request by blurbery**
## Problem

The next episode could start with audio but no picture after Next Up. Backing out and opening the episode again restored the picture. I could reproduce this when the countdown completed and when pressing Play Now. On iOS, a playing preview could turn black as it expanded.

This follows #221. Keeping the native player alive was part of the fix, but the SwiftUI video surface could still be destroyed and recreated between the preview and full-screen branches. Retained native state could also replay the previous episode's end-of-file event into the successor load.

## What I changed

- Keep one `AetherPlayerSurface` in the view hierarchy. Next Up changes its bounds, position and clipping instead of creating another video view. Expansion uses the existing picture and playback position, with a 0.3-second animation that respects Reduce Motion.
- If the requested episode is already playing, Play Now only expands the surface. It does not reload, seek or restart the engine.
- If the successor has not started, request one load and keep the Next Up presentation until that load reports its own first frame. Repeated actions while loading do not start more loads. Audio-only playback uses the playing milestone instead.
- Retire the previous load epoch before starting the next one. Ignore stale progress and completion events, and delay fetching the following candidate until the successor has a picture.
- Make the iOS Next Up preview smaller. Put the primary controls beside it in landscape/wide layouts and below it in portrait. Only the optional On Deck section scrolls; reaching Play Now should not require scrolling the whole screen.
- Preserve the existing tvOS focus structure and clip its retained preview to the scroll viewport. Keep error overlays, paused playback and the surface's full-screen accessibility exposure intact.

No server, database, authentication, deployment or unrelated navigation changes are included.

### Engine dependency

The AetherEngine pin moves from `e238d1d2da52b58bffd7f8135bb2b04e32915cee` to [`653be639441d3f7d06332a2f995950497ad62e0f`](https://github.com/blurbery/AetherEngine/commit/653be639441d3f7d06332a2f995950497ad62e0f). This stays on the existing 6.34.0-based patch line; it is not an unrelated engine version upgrade.

The handover patch retires the old observers and published EOF/time/readiness state without detaching the retained item and layer. Async native callbacks are fenced by their session. The native URL/HLS bypass now consumes the same pending in-place handover flag as the other native path. Normal stop still tears down playback; same-content recovery keeps its existing contract.

## Verification and measured regressions

I tested the installed iOS simulator app against a real server. Both paths now worked in manual testing:

- Countdown completed: the next episode showed a picture.
- Play Now pressed before the countdown completed: the next episode showed a picture and the transition worked smoothly.

These are simulator observations, not physical iPhone or Apple TV certification. Private server addresses, account information, media details and live-session captures are deliberately excluded from this PR.

### Before/after lifecycle measurements

These are event counts and identity assertions, not FPS or device-latency benchmarks.

| Check | Before / negative control | With the fix |
| --- | --- | --- |
| Preview/full-screen video host | Legacy two-branch harness replaces the actual Aether view | 40 transitions, 1 actual Aether view, 0 replacements |
| Expanding an already-playing native preview | Structural replacement reproduced separately | Same view, AVPlayer, AVPlayerItem and layer; 0 item changes; playback clock does not rewind |
| Native URL successor handover | 2 current-item changes, including 1 nil item | 1 item change, 0 nil items |
| Retained EOF state | Previous EOF replays into a new subscription | No stale EOF replay |
| Queued previous-item EOF callback | Old callback escapes the handover boundary | Callback is ignored |
| Repeated Next Up actions | Duplicate-load risk | 100 resolver calls produce 1 load decision |
| Paused preview resize | Covered by the new native fixture test | Stays paused, no item change or position reset |

The decoded-video test uses the existing bundled H.264/AAC fixture. It waits for an actual ready-for-display layer before expansion, then verifies the successor's first-frame latch and retained view/player/layer. The CI simulator's native URL route is selected explicitly so this test exercises AVPlayer rather than silently falling back to software decoding.

The mobile layout test measures the real production action panel at 568×320, 844×390, 390×844 and 1024×768 logical points, with a long optional shelf competing for space. All four keep the action panel within the requested viewport. Portrait preview width is capped at 260 points. This does not claim coverage of every Dynamic Type size or localization.

### CI results

Latest CI follow-up: [validation at `bfb9058`](https://github.com/Silo-Server/silo-apple/actions/runs/33715397674) passed. The complete iOS suite executed 1,174 tests with 1 existing skip and 0 failures, including both corrected navigation tests. Apple TV and macOS builds passed. The separate synthetic playback-recording test passed and the video was uploaded. CodeRabbit is green and its baseline-failure thread is resolved. The earlier results below are retained for comparison.

| Run | Result |
| --- | --- |
| [AetherEngine validation](https://github.com/blurbery/AetherEngine/actions/runs/33710488685) | 1,961 tests in 282 suites passed; iOS, tvOS, visionOS and macOS builds and transport probe passed |
| Engine negative controls in the same run | Each of the 3 new engine tests failed independently against the old `e238d1d…` production implementation and passed with the patch |
| [Silo application validation at `1a24cb4`](https://github.com/blurbery/silo-apple/actions/runs/33711266780) | iOS compiled; all 5 surface tests and the new completion-policy tests passed; tvOS and macOS builds passed |
| Silo complete iOS suite | 1,174 tests executed, 1 skipped, 3 existing assertion failures in 2 navigation tests; not a green full suite |
| [Unchanged upstream comparison](https://github.com/blurbery/silo-apple/actions/runs/33711266983) | Checked out exact upstream `8948ca88d8d765fbf15c147773d8af2d28b8f0ed`: 1,165 tests, 1 skipped, the same 3 assertion failures |
| [Evidence-capture run at `1577bc7`](https://github.com/blurbery/silo-apple/actions/runs/33712365506) | iOS compiled; all 5 surface tests passed; full suite again reported 1,174 tests, 1 skip and the same 3 baseline assertion failures; tvOS/macOS builds passed. I cancelled the run after the separate motion-recording step stalled; this is not a green workflow |

The earlier failures were three stale assertions in `UICustomizationPreferencesTests`: two in `testMainTabLibrarySnapshotRejectsPreviousProfileWithOverlappingId` and one in `testMainTabProjectionOnlyShowsCategoriesBackedByCurrentLibraries`. The application intentionally falls back to Home/Recommendations/Calendar when no library roots remain; `testMainTabProjectionFailsClosedAndFiltersUnavailableLibraries` already verifies that behavior. Commit `f9af6cc` aligns the stale expectations while preserving the checks that reject another profile's library, remove revoked pins, and return invalid selections to Home. Production navigation is unchanged. No tests are skipped or quarantined to bypass these failures; the complete suite still fails the workflow on any test failure.

The first follow-up run then failed before compilation because its hosted runner had no usable iPhone simulator destination. Commit `bfb9058` prepares a simulator matching the selected Xcode SDK and reuses its explicit ID for the build, tests, and recording. Missing runtimes are downloaded only when needed. Setup and recording each have a ten-minute limit. All of this validation runs on GitHub-hosted machines.

For reproducibility, the completed engine suite reported 63.273 seconds and the Silo suite at `1577bc7` reported 55.008 seconds of test execution (72.518 seconds elapsed). These are runner test timings, not app startup or transition performance claims.

## Evidence and remaining checks

- [Completed synthetic test results and logs](https://github.com/blurbery/silo-apple/actions/runs/33711266780/artifacts/9877278766) include the native lifecycle assertions and mobile geometry results.
- [Before/after screenshots, full-viewport layout images, test results and logs at `1577bc7`](https://github.com/blurbery/silo-apple/actions/runs/33712365506/artifacts/9877713041) are uploaded. I inspected the decoded test-pattern screenshots: the preview shows frame 4 at 0.167 seconds; the expanded surface shows frame 24 at 1.000 seconds. The accompanying test asserts the same view, player, item and layer across expansion. The images and identity assertions support continuity; still images alone do not prove the animation is smooth.
- The synthetic motion recording now completes successfully. The [latest iOS evidence artifact](https://github.com/Silo-Server/silo-apple/actions/runs/33715397674/artifacts/9878560188) contains `preview-transition.mp4`, the full-suite result bundle, and the separate passing playback-recording test result. Earlier failed capture attempts remain documented above as historical results.
- Evidence is uploaded through GitHub Actions, not committed as PR-only media assets. Validation artifacts have 7-day retention; the explicitly approved public-source simulator bundle has 3-day retention. Reviewers can rerun the workflow after expiry.
- Physical Apple TV/iPhone verification remains needed, especially HDR/Dolby Vision, HDMI refresh changes, PiP and AirPlay. The automated tests and native builds reduce regression risk but cannot prove every device/codec combination.

The playback app code has been saved and the two reported manual transition paths are confirmed working.

## AI disclosure

I orchestrated OpenAI Codex desktop using `gpt-5.6-sol` with `ultra` reasoning for the investigation, implementation, regression tests and this write-up. I directed the work and manually tested the countdown and Play Now paths in the simulator.

The CI follow-up was AI-assisted in Codex desktop using GitHub CLI, Git, `swiftc`, Ruby, Bash, and `jq`. This task did not expose its exact model identifier. Adversarial review compared the corrected expectations with the existing fallback tests and confirmed that stale/revoked library rejection and the full-suite failure gate remain intact; workflow YAML, shell syntax, and runtime-selection edge cases were checked locally.

## Summary by CodeRabbit

- **New Features**
  - Added responsive Next Up layouts for mobile, tvOS, and wider screens.
  - Keeps the video player visible and persistent while previewing, expanding, or transitioning to the next episode.
  - Added clearer Next Up actions, countdown controls, loading states, and accessibility support.

- **Bug Fixes**
  - Prevented the post-roll preview from disappearing while the next episode loads.
  - Improved autoplay and “Play Now” behavior when content is still preparing.
  - Reduced unnecessary player resets during episode changes.

### CODE REVIEW

File: .github/workflows/player-regression.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        if: matrix.action == 'test'
+        timeout-minutes: 12
+        run: |
+          set -o pipefail
+          xcodebuild test-without-building -project iosApp/Silo.xcodeproj -scheme Silo \
+            -destination 'platform=iOS Simulator,name=iPhone 17 Pro' -derivedDataPath /tmp/silo-derived \
+            -parallel-testing-enabled NO \
+            -test-timeouts-enabled YES -maximum-test-execution-time-allowance 60 \
+            -resultBundlePath /tmp/player-validation.xcresult \
+            CODE_SIGNING_ALLOWED=YES CODE_SIGN_IDENTITY=- | tee /tmp/player-validation.log
```
[the remark this answers is not in the document]
Comment (blurbery) [reply .github/workflows/player-regression.yml:126]: Fixed in f9af6cc by correcting the three stale navigation expectations while preserving the checks that reject stale and revoked libraries. The complete iOS suite now passes: 1,174 tests, 1 existing skip, and 0 failures. tvOS and macOS builds are green too: [passing CI run](https://github.com/Silo-Server/silo-apple/actions/runs/33715397674). No tests were quarantined or disabled, and the verified playback code is unchanged.
